### PR TITLE
Adopt kuery3

### PR DESCRIPTION
**Pull request by kilnerm**
This PR updates the ORM with the necessary changes to use SwiftKuery 3.0.

### CODE REVIEW

File: Sources/SwiftKueryORM/Database.swift [hunk 1]
```diff
@@ -35,6 +35,9 @@ public typealias ConnectionPoolOptions = SwiftKuery.ConnectionPoolOptions
 
 public class Database {
 
+    /// Definition of a databaseTask completion handler which accepts an optional Connection and optional Error
+    public typealias databaseTask = (Connection?, QueryError?) -> ()
```
[the remark this answers is not in the document]
Comment (kilnerm) [reply Sources/SwiftKueryORM/Database.swift]: Updated.

File: Sources/SwiftKueryORM/Database.swift [hunk 2]
```diff
@@ -64,17 +67,36 @@ public class Database {
         self.connectionStrategy = .pool(pool)
     }
 
-    /// Constructor for a custom connection generator
-    public init(generator: @escaping () -> Connection?) {
+    /**
+     Constructor for a custom connection generator
+     The generator must execute the supplied database task on a connected database Connection.
+
```
[the remark this answers is not in the document]
Comment (kilnerm) [reply Sources/SwiftKueryORM/Database.swift]: I expanded the comment and added similar comments to the other initializers.

File: Sources/SwiftKueryORM/Database.swift [hunk 3]
```diff
@@ -45,12 +48,29 @@ public class Database {
...
-        case generator(() -> Connection?)
+        case generator(((@escaping DatabaseTask)) -> ())
     }
 
     private let connectionStrategy: ConnectionStrategy
 
-    /// Constructor for a single connection which becomes a connection pool
+    /**
+     Create a Database instance which uses a single connection to perform each operation.
+     The connection is wrapped in a connection pool which prevents any concurrent operations being carried out on the connection
```
Comment (djones6) [inline Sources/SwiftKueryORM/Database.swift]: I don't think we need to describe this internal implementation detail.  From the user's perspective, they are just creating a connection to a database.  We should also mention the trade-offs of doing this vs. the pool / generator options.  Perhaps we could reword this block as follows:
> Create a Database instance which uses a single connection to perform each operation.  The connection will remain open for the lifetime of the Database.
> It is safe for you to issue concurrent operations on the database: the ORM will ensure those operations are executed sequentially.
> This connection strategy provides lower latency requests, as a new connection does not need to be established for each operation, and has a small resource overhead.  However, scalability is limited as all database operations are serialized.

File: Sources/SwiftKueryORM/Database.swift [hunk 4]
```diff
@@ -59,22 +79,50 @@ public class Database {
         self.init(singleConnectionPool)
     }
 
-    /// Default constructor for a connection pool
+    /**
+     Create a Database instance which uses a connection pool. A connection will be removed from the pool for each operation and returned to the pool nce the operation is complete.
+     The pooled connections will hold system resources while not in use but will result in more efficient processing of operations when the pool has an available connection.
```
Comment (djones6) [inline Sources/SwiftKueryORM/Database.swift]: Could we reword this similar to:
> Create a Database instance with multiple connections, managed by a connection pool, allowing operations to be performed concurrently.  These connections will remain open for the lifetime of the Database.
> This connection strategy provides lower latency requests, as a new connection does not need to be established for each operation.  You can choose between better performance or lower resource consumption by adjusting the number of connections in the pool.

File: Sources/SwiftKueryORM/Database.swift [hunk 5]
```diff
@@ -59,22 +79,50 @@ public class Database {
...
+     */
     public init(_ pool: ConnectionPool) {
         self.connectionStrategy = .pool(pool)
     }
 
-    /// Constructor for a custom connection generator
-    public init(generator: @escaping () -> Connection?) {
+    /**
+     Create a Database instance which uses short-lived connections that are generated on demand. A new Connection is created for every operation, and will be closed once the operation completes.
+     An advantage of this approach is that memory and resources are not tied up with connections when they are not required, however, the process of establishing a connection will impact the time taken to process each operation.
```
Comment (djones6) [inline Sources/SwiftKueryORM/Database.swift]: Could we change this line to be consistent with the changes above:
> This connection strategy allows for minimal resource consumption when idle, and allows multiple operations to be performed concurrently.  However, the process of establishing a connection will increase the time taken to process each operation.

File: Sources/SwiftKueryORM/Database.swift [hunk 6]
```diff
@@ -59,22 +79,50 @@ public class Database {
...
 
-    /// Default constructor for a connection pool
+    /**
+     Create a Database instance which uses a connection pool. A connection will be removed from the pool for each operation and returned to the pool nce the operation is complete.
+     The pooled connections will hold system resources while not in use but will result in more efficient processing of operations when the pool has an available connection.
+     Below is an example code which creates a connection pool and uses it to create a Database instance:
+     ```swift
+     let connectionPool = PostgreSQLConnection.createPool(host: host, port: port, options: [.userName("myUser"), .password("myPassword")], poolOptions: ConnectionPoolOptions(initialCapacity: 5, maxCapacity: 10))
+
+     let db = Database(pgresPool)
```
Comment (djones6) [inline Sources/SwiftKueryORM/Database.swift]: This should be `connectionPool` rather than `pgresPool`

**Pull request merged**